transport/tchannel/handler.go

```diff
@@ -267,14 +267,15 @@ func getSystemError(err error) error {
 	if _, ok := err.(tchannel.SystemError); ok {
 		return err
 	}
-	status := tchannel.ErrCodeUnexpected
-	code := yarpcerrors.FromError(err).Code()
-	if code == yarpcerrors.CodeInvalidArgument || code == yarpcerrors.CodeUnimplemented {
-		status = tchannel.ErrCodeBadRequest
-	} else if code == yarpcerrors.CodeDeadlineExceeded {
-		status = tchannel.ErrCodeTimeout
+	if !yarpcerrors.IsStatus(err) {
```

Comment:
You could simplify this whole change to:
```
status := yarpcerrors.FromError(err)
tchannelCode := _codeToTChannelCode[status.Code()]
return tchannel.NewSystemError(tchannelCode, status.Message()
```
since `yarpcerrors.FromError()` returns an `CodeUnknown` error if it's not valid (which maps to `tchannel.ErrCodeUnexpected`)

Reply:
_codeToTChannelCode might have a miss, still need the ok check.